CMakeLists.txt

```diff
@@ -0,0 +1,91 @@
+cmake_minimum_required (VERSION 2.6)
+project (LuaGumbo C)
+
+set(CMAKE_C_STANDARD_REQUIRED true)
+set(CMAKE_C_STANDARD 99)
+
+if(UNIX)
+  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+endif()
+
+# Lua
+if(WIN32)
+  find_package (Lua REQUIRED)
+else()
+  # Only the include dir matters on UNIX
+  find_package (Lua)
+  if(NOT LUA_INCLUDE_DIR)
+    message( FATAL_ERROR "Variable `LUA_INCLUDE_DIR' expected to be defined.")
+  endif()
+endif()
+
+# Gumbo
+set(LIB_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/lib)
+set(WINDOWS_LIB_INCLUDE_DIR ${LIB_INCLUDE_DIR}/visualc/include)
+set(LIBGUMBO_OBJ attribute char_ref error parser string_buffer string_piece svg_attrs svg_tags tag tag_lookup tokenizer utf8 util vector)
+
+add_library( attribute lib/attribute.c lib/attribute.h lib/gumbo.h lib/util.h )
```

Comment:
`add_library` seems to create a library archive file per compilation unit instead of object files. Is that a cmake thing or...?

Reply:
Looks like we want to use `add_library( name OBJECT files )` instead; will make that change.

Reply:
Scratch that, `OBJECT` types can only be included as sources for other `add_library` calls. Will look into this more and figure out a better solution.

Reply:
Ok, figured out how to use `OBJECT` properly. Now only object files should be created for the libgumbo parts.